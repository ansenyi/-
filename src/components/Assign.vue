<template>
  <div class="assign center">
    <div class="one-column left-column">
      <h4 class="box-title">{{ leftTitle }}</h4>
      <div class="box-content">
        <div class="search-input">
          <input type="text" placeholder="模糊搜索" v-model="leftSearchInput" @keyup="leftSearchLike" autocomplete="off" />
          <i class="input-cancel icon-cancel" v-show="showCancel" @click="leftSearchInput = '';leftSearchLike()"></i>
        </div>
        <div class="my-bar" v-bar>
          <ul ref="leftList" @click="leftClick($event)" @change="leftChange($event)">
            <li v-for="(items, index) in leftList" v-show="leftLikeList.indexOf(items.id) !== -1" :key="index" :data-index="index" :data-id="items.id">
              <p :class="chosenLeft === String(items.id) ? 'chosen' : ''">
                <i class="icon-right" v-show="actives.indexOf(String(index)) === -1"></i>
                <i class="icon-down" v-show="actives.indexOf(String(index)) !== -1"></i>
                {{ items.name }}
              </p>
              <ul v-show="actives.indexOf(String(index)) !== -1" :data-index="index">
                <li v-for="(item, index) in items.children" :key="index" :data-id="item.id" :data-name="item.name">
                  <input type="checkbox" v-model="item.checked" :id="'l' + items.id + item.id" :value="index" />
                  <label :for="'l' + items.id + item.id">{{ item.name }}</label>
                </li>
              </ul>
            </li>
          </ul>
        </div>
        <!--<div class="replace-little-bar" v-show="!chosenLeftList.length"></div>-->
        <div class="little-bar" v-bar>
          <div class="chosen-options">
            <div class="chosen-option" v-for="(item, index) in chosenLeftList" :key="index">
              {{ item.name }}
              <i class="chosen-cancel icon-cancel" @click="cancelChosenLeft(item.id, item.parentId, index)"></i>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="buttons">
      <button class="button" @click="toUnassigned()"><i class="icon-to-right"></i></button>
      <button class="button" @click="toAssigned()"><i class="icon-to-left"></i></button>
    </div>
    <div class="one-column right-column">
      <h4 class="box-title">{{ rightTitle }}</h4>
      <div class="box-content">
        <p @click="selectAll" v-show="select">全选</p>
        <p @click="unselectAll" v-show="!select">全不选</p>
        <div class="search-input">
          <input type="text" placeholder="模糊搜索" v-model="rightSearchInput" @keyup="rightSearchLike" autocomplete="off" />
          <i class="input-cancel icon-cancel" v-show="showCancelRight" @click="rightSearchInput = '';rightSearchLike()"></i>
        </div>
        <div class="my-bar" v-bar>
          <ul ref="rightList" @change="rightChange($event)">
            <li v-for="(item, index) in rightList" v-show="rightLikeList.indexOf(item.id) !== -1" :data-id="item.id" :data-name="item.name" :key="index">
              <input type="checkbox" v-model="item.checked" :id="'r' + item.id" :value="index" />
              <label :for="'r' + item.id">{{ item.name }}</label>
            </li>
          </ul>
        </div>
        <!--<div class="replace-little-bar" v-show="!chosenRightList.length"></div>-->
        <div class="little-bar" v-bar>
          <div class="chosen-options">
            <div class="chosen-option" v-for="(item, index) in chosenRightList" :key="index">
              {{ item.name }}
              <i class="chosen-cancel icon-cancel" @click="cancelChosenRight(item.id, index)"></i>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
  import { searchLike, _debounce } from '../common/js/util'
  export default {
    name: 'assign',
    data () {
      return {
        leftList: [],  // id, name, checked, children:{id, name, checked}
        leftLikeList: [],  // id
        rightList: [],  // id, name, checked
        rightLikeList: [],  // id
        saveList: [],  // userId, orgId, editType
        chosenLeftList: [],  // 已选要解除分配的人员的'userId', userName, orgId
        chosenRightList: [],  // 已选要分配的人员的'userId', userName
        leftSearchInput: '',
        rightSearchInput: '',
        actives: [], // 显示出成员的部门的index
        chosenLeft: '', // 已选要分配人员的部门的orgId
        toAssigned: _debounce(this.toAssignedHandler),
        toUnassigned: _debounce(this.toUnassignedHandler)
      }
    },
    props: {
      leftTitle: {
        type: String,
        default: ''
      },
      rightTitle: {
        type: String,
        default: ''
      },
      initPost: {
        type: String,
        default: ''
      },
      leftPre: {
        type: String,
        default: ''
      },
      rightPre: {
        type: String,
        default: ''
      },
      queryLeft: {
        type: String,
        default: ''
      },
      queryRight: {
        type: String,
        default: ''
      },
      childrenPost: {
        type: String,
        default: ''
      },
      leftMsg: {
        type: String,
        default: ''
      },
      theLeftMsg: {
        type: String,
        default: ''
      },
      rightMsg: {
        type: String,
        default: ''
      },
      savePost: {
        type: String,
        default: ''
      }
    },
    computed: {
      select () {
        return this.rightList.length === 0 || this.chosenRightList.length !== this.rightList.length
      },
      showCancel () {
        return this.leftSearchInput !== ''
      },
      showCancelRight () {
        return this.rightSearchInput !== ''
      }
    },
    created () {
      if (this.$route.name === 'RoleRightAssign') {
        this.$http.post(this.queryLeft, JSON.stringify({
          like: ''
        })).then((res) => {
          if (res.data.status) {
            console.log('左侧模糊查询后的数据：')
            console.log(res)
            res.data.data.forEach((item) => {
              let suf = ''
              switch (item.roleType) {
                case 1 : suf = '（默认）'
                  break
                case 2 : suf = '（管理员）'
              }
              this.leftList.push({id: item[this.leftPre + 'Id'], name: item[this.leftPre + 'Name'] + suf})
              this.leftLikeList.push(item[this.leftPre + 'Id'])
            })
          }
        }).catch(err => console.log(err))
        this.$http.post(this.queryRight, JSON.stringify({
          like: ''
        })).then((res) => {
          if (res.data.status) {
            console.log('右侧模糊查询后的数据：')
            console.log(res)
            res.data.data.forEach((item) => {
              this.rightList.push({id: item[this.rightPre + 'Id'], name: item[this.rightPre + 'Name'], checked: false})
              this.rightLikeList.push(item[this.rightPre + 'Id'])
            })
            if (this.$route.params.deviceName) {
              this.rightSearchInput = this.$route.params.deviceName
              this.rightSearchLike()
            }
          }
        }).catch(err => console.log(err))
      } else {
        this.$http.post(this.initPost, '').then((res) => {
          if (res.data.status) {
            if (this.$route.name === 'RoleAssign') {
              res.data.data[1].forEach((item) => {
                let suf = ''
                switch (item.roleType) {
                  case 1 : suf = '（默认）'
                    break
                  case 2 : suf = '（管理员）'
                }
                this.leftList.push({id: item[this.leftPre + 'Id'], name: item[this.leftPre + 'Name'] + suf})
                this.leftLikeList.push(item[this.leftPre + 'Id'])
              })
              res.data.data[0].forEach((item) => {
                this.rightList.push({id: item[this.rightPre + 'Id'], name: item[this.rightPre + 'Name'], checked: false})
                this.rightLikeList.push(item[this.rightPre + 'Id'])
              })
            } else {
              res.data.data[0].forEach((item) => {
                this.leftList.push({id: item[this.leftPre + 'Id'], name: item[this.leftPre + 'Name']})
                this.leftLikeList.push(item[this.leftPre + 'Id'])
              })
              res.data.data[1].forEach((item) => {
                this.rightList.push({id: item[this.rightPre + 'Id'], name: item[this.rightPre + 'Name'], checked: false})
                this.rightLikeList.push(item[this.rightPre + 'Id'])
              })
            }
          }
        }).catch(err => console.log(err))
      }
    },
    methods: {
      async leftSearchLike () {
        let property = this.leftPre + 'Id'
        let path = this.queryLeft
        let option
        if (this.$route.name === 'PersonDeptManage') {
          option = {level: 2, like: this.leftSearchInput}
        } else {
          option = {like: this.leftSearchInput}
        }
        this.leftLikeList = []
        this.leftLikeList = await searchLike(this.leftSearchInput, this.leftList, this.leftLikeList, property, path, option)
      },
      async rightSearchLike () {
        let property = this.rightPre + 'Id'
        let path = this.queryRight
        let option = {like: this.rightSearchInput}
        this.rightLikeList = []
        this.rightLikeList = await searchLike(this.rightSearchInput, this.rightList, this.rightLikeList, property, path, option)
      },
      // 已选标签的添加
      leftChange (e) {
        let target = e ? e.target : window.event.srcElement
        let node = target.parentNode
        if (target.tagName.toLowerCase() === 'input') {
          let index = target.value
          let parentIndex = node.parentNode.dataset.index
          this.leftList[parentIndex].children[index].checked ? this.chosenLeftList.push({ id: node.dataset.id, name: node.dataset.name, parentId: node.parentNode.parentNode.dataset.id })
            : this.chosenLeftList.splice(this.chosenLeftList.findIndex((item) => item.id === node.dataset.id), 1)
        }
      },
      // 子列表的显示
      leftClick (e) {
        let target = e ? e.target : window.event.srcElement
        let node = target.parentNode
        let dataset
        switch (target.tagName.toLowerCase()) {
          case 'i' : dataset = node.parentNode.dataset
            break
          case 'p' : dataset = node.dataset
            break
          default :
            return
        }
        console.log('this.actives的数据：')
        console.log(this.actives)
        this.chosenLeft = dataset.id
        if (this.actives.indexOf(dataset.index) !== -1) {
          let index = this.actives.indexOf(dataset.index)
          this.actives.splice(index, 1)
          console.log(this.actives)
          return
        }
        this.getChildren(dataset.index, dataset.id)
        this.actives.push(dataset.index)
      },
      getChildren (index, id) {
        this.$http.post(this.childrenPost, JSON.stringify({
          [this.leftPre + 'Id']: id
        })).then((res) => {
          console.log('取子列表数据后的数据：')
          console.log(res.data.data)
          if (res.data.status) {
            let children = []
            res.data.data.forEach((item) => {
              let checked = this.$route.name !== 'RoleRightAssign'
                ? this.chosenLeftList.findIndex(i => i.id === String(item[this.rightPre + 'Id'])) > -1
                : this.chosenLeftList.findIndex(i => i.id === String(item[this.rightPre + 'Id']) && i.parentId === String(id)) > -1
              children.push({id: item[this.rightPre + 'Id'], name: item[this.rightPre + 'Name'], checked: checked})
            })
            this.$set(this.leftList[index], 'children', children)
//            callback()
          }
        }).catch(err => console.log(err))
      },
      rightChange (e) {
        console.log('向右点击')
        let target = e ? e.target : window.event.srcElement
        if (target.tagName.toLowerCase() === 'input') {
          let node = target.parentNode
          let index = target.value
          // this.rightList[index].checked = !this.rightList[index].checked
          this.rightList[index].checked ? this.chosenRightList.push({id: node.dataset.id, name: node.dataset.name})
            : this.chosenRightList.splice(this.chosenRightList.findIndex(item => item.id === node.dataset.id), 1)
        }
      },
      toUnassignedHandler () {
        if (this.chosenLeftList.length === 0) {
          this.$emit('alertMsg', this.leftMsg)
          return
        }
        this.saveList = []
        this.chosenLeftList.forEach((item) => {
          this.saveList.push({[this.rightPre + 'Id']: item.id, [this.leftPre + 'Id']: item.parentId, editType: 2})
        })
        this.saveAll(() => {
//          let children = []
          let leftParents = [] // {index, id}
          this.chosenLeftList.forEach((item) => {
            let index = this.leftList.findIndex(i => String(i.id) === item.parentId)
            let child = this.leftList[index].children.find(i => String(i.id) === item.id)
            child.checked = false
            console.log('index')
            console.log(index)
            // this.getChildren(index, item.parentId)
            leftParents.findIndex(p => p.id === item.parentId) < 0 ? leftParents.push({index: index, id: item.parentId}) : null
            if (this.$route.name !== 'RoleRightAssign') {
//              children.push(child)
              this.rightList.push(child)
            }
          })
          leftParents.forEach(p => this.getChildren(p.index, p.id))
//          this.rightList = children.concat(this.rightList)
          if (this.$route.name !== 'RoleRightAssign') {
            this.rightList.sort((x, y) => {
              return x.id - y.id
            })
          }
          this.chosenLeftList = []
          this.rightSearchLike()
          this.leftSearchLike()
          this.$emit('alertMsg', '解除分配成功！')
        })
      },
      toAssignedHandler () {
        if (this.chosenLeft === '') {
          this.$emit('alertMsg', this.theLeftMsg)
          return
        }
        if (this.chosenRightList.length === 0) {
          this.$emit('alertMsg', this.rightMsg)
          return
        }
        this.saveList = []
        this.chosenRightList.forEach((item) => {
          this.saveList.push({[this.rightPre + 'Id']: item.id, [this.leftPre + 'Id']: this.chosenLeft, editType: 1})
        })
        this.saveAll(() => {
          this.chosenRightList.forEach((item) => {
            let rIndex = this.rightList.findIndex(i => String(i.id) === item.id)
            this.rightList[rIndex].checked = false
            if (this.$route.name !== 'RoleRightAssign') {
              this.rightList.splice(rIndex, 1)
            }
          })
          let lIndex = this.leftList.findIndex(i => String(i.id) === this.chosenLeft)
          this.getChildren(lIndex, this.chosenLeft)
          this.chosenRightList = []
          this.leftSearchLike()
          this.rightSearchLike()
          this.$emit('alertMsg', '分配成功！')
        })
      },
      saveAll (callback) {
        if (this.saveList.length > 0) {
          this.$http.post(this.savePost, JSON.stringify(this.saveList))
            .then((res) => {
              if (res.data.status) {
                this.saveList = []
                callback()
              } else {
                this.saveList = []
                this.$emit('alertMsg', res.data.msg)
              }
            }).catch((err) => {
              console.log(err)
              this.saveList = []
              this.$emit('alertMsg', '保存失败，请重试。')
            })
        }
      },
      cancelChosenLeft (id, parentId, index) {
        let children = this.leftList.find(i => String(i.id) === parentId).children
        let left = children.find((i) => {
          return String(i.id) === id
        })
        console.log('left的内容是：')
        console.log(left)
        if (!left) {
          console.log('左边选中的不在左边')
          return
        }
        left.checked = false
        this.chosenLeftList.splice(index, 1)
      },
      cancelChosenRight (id, index) {
        let right = this.rightList.find((i) => {
          return String(i.id) === id
        })
        if (!right) {
          console.log('右边选中的不在右边')
          return
        }
        right.checked = false
        this.chosenRightList.splice(index, 1)
      },
      selectAll () {
        if (this.rightLikeList.length === this.rightList.length) {
          this.chosenRightList = []
          this.rightList.forEach((d) => {
            d.checked = true
            this.chosenRightList.push({id: String(d.id), name: d.name})
          })
        } else {
          this.rightLikeList.forEach((id) => {
            let right = this.rightList.find(d => d.id === id)
            if (!this.chosenRightList.some(d => d.id === String(id))) {
              right.checked = true
              this.chosenRightList.push({id: String(right.id), name: right.name})
            }
          })
        }
      },
      unselectAll () {
        this.chosenRightList = []
        this.rightList.forEach((d) => {
          d.checked = false
        })
      }
    }
  }
</script>

<style lang="stylus" scoped>
  @import '../common/stylus/mixin.styl'
  .assign
    three-columns()
    padding-top 18vh
    .replace-little-bar
      height 10px
    .right-column
      .box-content
        p
          margin -16px 0 4px
          text-align right
          cursor pointer
          color #2dc5fa
          font-size 14px
    .button
     background-color #0077d5
</style>
