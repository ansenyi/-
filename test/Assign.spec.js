/* eslint-disable no-undef,no-unused-vars */
import 'babel-polyfill'
import jest from 'jest-mock'
import { mount, createLocalVue } from 'vue-test-utils'
import Assign from '@/components/Assign.vue'
import RoleAssign from '@/components/Role/RoleAssign.vue'
import RoleRightAssign from '@/components/Role/RoleRightAssign.vue'
import PersonDeptManage from '@/components/person/PersonDeptManage.vue'
import moxios from 'moxios'
import axios from 'axios'
import vueAxios from 'vue-axios'
import './utils/proxy-dataset'

describe('assign', () => {
  const localVue = createLocalVue()
  localVue.use(vueAxios, axios)
  function initAssign (page, routeName, data) {
    const $route = {
      name: routeName,
      params: {
        deviceName: ''
      }
    }
    const cmp = mount(page, {
      localVue,
      mocks: {
        $route
      },
      data: data
    })
    return cmp
  }
  describe('role assign', () => {
    it('mock created', async() => {
      moxios.install()
      moxios.stubRequest('roleUserManagePage', {
        status: 200,
        response: {
          status: true,
          data: [[{userId: 2, userName: 'b'}], [{roleId: 1, roleName: 'a'}]]
        }
      })
      const cmp = initAssign(RoleAssign, 'RoleAssign')
      const assign = cmp.find(Assign)
      await moxios.wait()
      await assign.vm.$nextTick()
      await cmp.vm.$nextTick()
      expect(assign.element).toMatchSnapshot()
      expect(assign.vm.leftList).toHaveLength(1)
      moxios.uninstall()
    })
    it('left click and to unassigned', async() => {
      moxios.install()
      moxios.stubRequest('roleUserManagePage', {
        status: 200,
        response: {
          status: true,
          data: [[{userId: 2, userName: 'b'}], [{roleId: 1, roleName: 'a'}]]
        }
      })
      const cmp = initAssign(RoleAssign, 'RoleAssign')
      const assign = cmp.find(Assign)
      await moxios.wait()
      await assign.vm.$nextTick()
      await cmp.vm.$nextTick()
      assign.vm.actives = ['0']
      assign.find('.left-column').find('.icon-right').trigger('click')
      expect(assign.vm.chosenLeft).toBe('1')
      expect(assign.vm.actives).toHaveLength(0)
      assign.vm.actives = ['0']
      assign.find('.left-column').find('p').trigger('click')
      expect(assign.vm.chosenLeft).toBe('1')
      expect(assign.vm.actives).toHaveLength(0)
      assign.vm.leftList = [{id: 1, name: 'a', children: [{id: 11, name: 'aa'}]}]
      assign.vm.actives = ['0']
      await assign.vm.$nextTick()
      assign.find('.left-column').find('input[type=checkbox]').trigger('click')
      await assign.vm.$nextTick()
      expect(assign.find('.left-column').contains('.chosen-option')).toBeTruthy()
      console.log('left click end')
      moxios.stubRequest('editUserInRole', {
        status: 200,
        response: {
          status: true
        }
      })
      assign.find('.buttons .button').trigger('click')
      await moxios.wait()
      await assign.vm.$nextTick()
      expect(assign.vm.leftList[0].children).toHaveLength(0)
      expect(assign.vm.rightList[1].id).toBe('11')
      console.log('unassigned end')
      moxios.uninstall()
    })
    it('to assigned', async() => {
      moxios.install()
      moxios.stubRequest('roleUserManagePage', {
        status: 200,
        response: {
          status: true,
          data: [[{userId: 2, userName: 'b'}], [{roleId: 1, roleName: 'a'}]]
        }
      })
      const cmp = initAssign(RoleAssign, 'RoleAssign')
      const assign = cmp.find(Assign)
      await moxios.wait()
      await assign.vm.$nextTick()
      await cmp.vm.$nextTick()
      assign.vm.actives = ['0']
      assign.vm.chosenLeft = '1'
      await assign.vm.$nextTick()
      assign.find('.right-column').find('input[type=checkbox]').trigger('click')
      await assign.vm.$nextTick()
      expect(assign.find('.right-column').contains('.chosen-option')).toBeTruthy()
      console.log('right click end')
      moxios.stubRequest('editUserInRole', {
        status: 200,
        response: {
          status: true
        }
      })
      assign.find('.buttons .button:last-of-type').trigger('click')
      await moxios.wait()
      await assign.vm.$nextTick()
      expect(assign.vm.rightList).toHaveLength(0)
      console.log('assigned end')
      moxios.uninstall()
    })
  })
  describe('role right assign', () => {
    it('init', async() => {
      moxios.install()
      moxios.stubRequest('roleDeviceManagePage', {
        status: 200,
        response: {
          status: true,
          data: [[{roleId: 1, roleName: 'a'}], [{deviceId: 2, deviceName: 'b'}]]
        }
      })
      const cmp = initAssign(RoleRightAssign, 'RoleRightAssign')
      const assign = cmp.find(Assign)
      await moxios.wait()
      await assign.vm.$nextTick()
      await cmp.vm.$nextTick()
      expect(cmp.element).toMatchSnapshot()
      moxios.uninstall()
    })
  })
  describe('dept manage', () => {
    it('init', async() => {
      moxios.install()
      moxios.stubRequest('orgManagePage', {
        status: 200,
        response: {
          status: true,
          data: [[{orgId: 1, orgName: 'a'}], [{userId: 2, userName: 'b'}]]
        }
      })
      const cmp = initAssign(PersonDeptManage, 'PersonDeptManage')
      const assign = cmp.find(Assign)
      await moxios.wait()
      await assign.vm.$nextTick()
      await cmp.vm.$nextTick()
      expect(cmp.element).toMatchSnapshot()
      moxios.uninstall()
    })
  })
})
