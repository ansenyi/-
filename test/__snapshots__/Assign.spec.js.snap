// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assign dept manage init 1`] = `
<div
  class="assign center"
>
  <div
    class="one-column left-column"
  >
    <h4
      class="box-title"
    >
      已有部门
    </h4>
     
    <div
      class="box-content"
    >
      <div
        class="search-input"
      >
        <input
          autocomplete="off"
          placeholder="模糊搜索"
          type="text"
        />
         
        <i
          class="input-cancel icon-cancel"
          style="display: none;"
        />
      </div>
       
      <div
        class="little-bar"
      >
        <div
          class="chosen-options"
        />
      </div>
       
      <div
        class="my-bar"
      >
        <ul>
          <li
            data-id="1"
            data-index="0"
          >
            <p
              class=""
            >
              <i
                class="icon-right"
              />
               
              <i
                class="icon-down"
                style="display: none;"
              />
              
              a1
            
            </p>
             
            <ul
              style="display: none;"
            />
          </li>
        </ul>
      </div>
    </div>
  </div>
   
  <div
    class="buttons"
  >
    <button
      class="button"
    >
      <i
        class="icon-to-right"
      />
    </button>
     
    <button
      class="button"
    >
      <i
        class="icon-to-left"
      />
    </button>
  </div>
   
  <div
    class="one-column right-column"
  >
    <h4
      class="box-title"
    >
      未分配人员
    </h4>
     
    <div
      class="box-content"
    >
      <div
        class="search-input"
      >
        <input
          autocomplete="off"
          placeholder="模糊搜索"
          type="text"
        />
         
        <i
          class="input-cancel icon-cancel"
          style="display: none;"
        />
      </div>
       
      <div
        class="little-bar"
      >
        <div
          class="chosen-options"
        />
      </div>
       
      <div
        class="my-bar"
      >
        <ul>
          <li
            data-id="2"
            data-name="b"
          >
            <input
              id="2"
              type="checkbox"
            />
             
            <label
              for="2"
            >
              b2
            </label>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`assign role assign left click p show 1`] = `
<div
  class="assign center"
>
  <div
    class="one-column left-column"
  >
    <h4
      class="box-title"
    >
      已有角色
    </h4>
     
    <div
      class="box-content"
    >
      <div
        class="search-input"
      >
        <input
          autocomplete="off"
          placeholder="模糊搜索"
          type="text"
        />
         
        <i
          class="input-cancel icon-cancel"
          style="display: none;"
        />
      </div>
       
      <div
        class="little-bar"
      >
        <div
          class="chosen-options"
        />
      </div>
       
      <div
        class="my-bar"
      >
        <ul>
          <li
            data-id="1"
            data-index="0"
          >
            <p
              class=""
            >
              <i
                class="icon-right"
              />
               
              <i
                class="icon-down"
                style="display: none;"
              />
              
              a1
            
            </p>
             
            <ul
              style="display: none;"
            />
          </li>
          <li
            data-id="2"
            data-index="1"
          >
            <p
              class=""
            >
              <i
                class="icon-right"
              />
               
              <i
                class="icon-down"
                style="display: none;"
              />
              
              b2
            
            </p>
             
            <ul
              style="display: none;"
            />
          </li>
        </ul>
      </div>
    </div>
  </div>
   
  <div
    class="buttons"
  >
    <button
      class="button"
    >
      <i
        class="icon-to-right"
      />
    </button>
     
    <button
      class="button"
    >
      <i
        class="icon-to-left"
      />
    </button>
  </div>
   
  <div
    class="one-column right-column"
  >
    <h4
      class="box-title"
    >
      人员列表
    </h4>
     
    <div
      class="box-content"
    >
      <div
        class="search-input"
      >
        <input
          autocomplete="off"
          placeholder="模糊搜索"
          type="text"
        />
         
        <i
          class="input-cancel icon-cancel"
          style="display: none;"
        />
      </div>
       
      <div
        class="little-bar"
      >
        <div
          class="chosen-options"
        >
          <div
            class="chosen-option"
          >
            
            a
            
            <i
              class="chosen-cancel icon-cancel"
            />
          </div>
        </div>
      </div>
       
      <div
        class="my-bar"
      >
        <ul>
          <li
            data-id="2"
            data-name="b"
          >
            <input
              id="2"
              type="checkbox"
            />
             
            <label
              for="2"
            >
              b2
            </label>
          </li>
          <li
            data-id="1"
            data-name="a"
          >
            <input
              id="1"
              type="checkbox"
            />
             
            <label
              for="1"
            >
              a1
            </label>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`assign role assign mock created 1`] = `
<div
  class="assign center"
>
  <div
    class="one-column left-column"
  >
    <h4
      class="box-title"
    >
      已有角色
    </h4>
     
    <div
      class="box-content"
    >
      <div
        class="search-input"
      >
        <input
          autocomplete="off"
          placeholder="模糊搜索"
          type="text"
        />
         
        <i
          class="input-cancel icon-cancel"
          style="display: none;"
        />
      </div>
       
      <div
        class="little-bar"
      >
        <div
          class="chosen-options"
        />
      </div>
       
      <div
        class="my-bar"
      >
        <ul>
          <li
            data-id="1"
            data-index="0"
          >
            <p
              class=""
            >
              <i
                class="icon-right"
              />
               
              <i
                class="icon-down"
                style="display: none;"
              />
              
              a1
            
            </p>
             
            <ul
              style="display: none;"
            />
          </li>
        </ul>
      </div>
    </div>
  </div>
   
  <div
    class="buttons"
  >
    <button
      class="button"
    >
      <i
        class="icon-to-right"
      />
    </button>
     
    <button
      class="button"
    >
      <i
        class="icon-to-left"
      />
    </button>
  </div>
   
  <div
    class="one-column right-column"
  >
    <h4
      class="box-title"
    >
      人员列表
    </h4>
     
    <div
      class="box-content"
    >
      <div
        class="search-input"
      >
        <input
          autocomplete="off"
          placeholder="模糊搜索"
          type="text"
        />
         
        <i
          class="input-cancel icon-cancel"
          style="display: none;"
        />
      </div>
       
      <div
        class="little-bar"
      >
        <div
          class="chosen-options"
        />
      </div>
       
      <div
        class="my-bar"
      >
        <ul>
          <li
            data-id="2"
            data-name="b"
          >
            <input
              id="2"
              type="checkbox"
            />
             
            <label
              for="2"
            >
              b2
            </label>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`assign role assign mock created 2`] = `
<div
  class="assign center"
>
  <div
    class="one-column left-column"
  >
    <h4
      class="box-title"
    >
      已有角色
    </h4>
     
    <div
      class="box-content"
    >
      <div
        class="search-input"
      >
        <input
          autocomplete="off"
          placeholder="模糊搜索"
          type="text"
        />
         
        <i
          class="input-cancel icon-cancel"
          style="display: none;"
        />
      </div>
       
      <div
        class="little-bar"
      >
        <div
          class="chosen-options"
        />
      </div>
       
      <div
        class="my-bar"
      >
        <ul>
          <li
            data-id="1"
            data-index="0"
          >
            <p
              class=""
            >
              <i
                class="icon-right"
              />
               
              <i
                class="icon-down"
                style="display: none;"
              />
              
              a1
            
            </p>
             
            <ul
              style="display: none;"
            />
          </li>
        </ul>
      </div>
    </div>
  </div>
   
  <div
    class="buttons"
  >
    <button
      class="button"
    >
      <i
        class="icon-to-right"
      />
    </button>
     
    <button
      class="button"
    >
      <i
        class="icon-to-left"
      />
    </button>
  </div>
   
  <div
    class="one-column right-column"
  >
    <h4
      class="box-title"
    >
      人员列表
    </h4>
     
    <div
      class="box-content"
    >
      <div
        class="search-input"
      >
        <input
          autocomplete="off"
          placeholder="模糊搜索"
          type="text"
        />
         
        <i
          class="input-cancel icon-cancel"
          style="display: none;"
        />
      </div>
       
      <div
        class="little-bar"
      >
        <div
          class="chosen-options"
        />
      </div>
       
      <div
        class="my-bar"
      >
        <ul>
          <li
            data-id="2"
            data-name="b"
          >
            <input
              id="2"
              type="checkbox"
            />
             
            <label
              for="2"
            >
              b2
            </label>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`assign role assign to unassigned 1`] = `
<div
  class="assign center"
>
  <div
    class="one-column left-column"
  >
    <h4
      class="box-title"
    >
      已有角色
    </h4>
     
    <div
      class="box-content"
    >
      <div
        class="search-input"
      >
        <input
          autocomplete="off"
          placeholder="模糊搜索"
          type="text"
        />
         
        <i
          class="input-cancel icon-cancel"
          style="display: none;"
        />
      </div>
       
      <div
        class="little-bar"
      >
        <div
          class="chosen-options"
        />
      </div>
       
      <div
        class="my-bar"
      >
        <ul>
          <li
            data-id="1"
            data-index="0"
          >
            <p
              class=""
            >
              <i
                class="icon-right"
              />
               
              <i
                class="icon-down"
                style="display: none;"
              />
              
              a1
            
            </p>
             
            <ul
              style="display: none;"
            >
              <li
                data-id="11"
                data-name="aa"
              >
                <input
                  id="12"
                  type="checkbox"
                />
                 
                <label
                  for="12"
                >
                  aa11
                </label>
              </li>
            </ul>
          </li>
          <li
            data-id="2"
            data-index="1"
          >
            <p
              class=""
            >
              <i
                class="icon-right"
              />
               
              <i
                class="icon-down"
                style="display: none;"
              />
              
              b2
            
            </p>
             
            <ul
              style="display: none;"
            />
          </li>
        </ul>
      </div>
    </div>
  </div>
   
  <div
    class="buttons"
  >
    <button
      class="button"
    >
      <i
        class="icon-to-right"
      />
    </button>
     
    <button
      class="button"
    >
      <i
        class="icon-to-left"
      />
    </button>
  </div>
   
  <div
    class="one-column right-column"
  >
    <h4
      class="box-title"
    >
      人员列表
    </h4>
     
    <div
      class="box-content"
    >
      <div
        class="search-input"
      >
        <input
          autocomplete="off"
          placeholder="模糊搜索"
          type="text"
        />
         
        <i
          class="input-cancel icon-cancel"
          style="display: none;"
        />
      </div>
       
      <div
        class="little-bar"
      >
        <div
          class="chosen-options"
        >
          <div
            class="chosen-option"
          >
            
            a
            
            <i
              class="chosen-cancel icon-cancel"
            />
          </div>
        </div>
      </div>
       
      <div
        class="my-bar"
      >
        <ul>
          <li
            data-id="2"
            data-name="b"
          >
            <input
              id="2"
              type="checkbox"
            />
             
            <label
              for="2"
            >
              b2
            </label>
          </li>
          <li
            data-id="1"
            data-name="a"
          >
            <input
              id="1"
              type="checkbox"
            />
             
            <label
              for="1"
            >
              a1
            </label>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`assign role right assign init 1`] = `
<div
  class="assign center"
>
  <div
    class="one-column left-column"
  >
    <h4
      class="box-title"
    >
      已有角色
    </h4>
     
    <div
      class="box-content"
    >
      <div
        class="search-input"
      >
        <input
          autocomplete="off"
          placeholder="模糊搜索"
          type="text"
        />
         
        <i
          class="input-cancel icon-cancel"
          style="display: none;"
        />
      </div>
       
      <div
        class="little-bar"
      >
        <div
          class="chosen-options"
        />
      </div>
       
      <div
        class="my-bar"
      >
        <ul>
          <li
            data-id="1"
            data-index="0"
          >
            <p
              class=""
            >
              <i
                class="icon-right"
              />
               
              <i
                class="icon-down"
                style="display: none;"
              />
              
              a1
            
            </p>
             
            <ul
              style="display: none;"
            />
          </li>
        </ul>
      </div>
    </div>
  </div>
   
  <div
    class="buttons"
  >
    <button
      class="button"
    >
      <i
        class="icon-to-right"
      />
    </button>
     
    <button
      class="button"
    >
      <i
        class="icon-to-left"
      />
    </button>
  </div>
   
  <div
    class="one-column right-column"
  >
    <h4
      class="box-title"
    >
      设备列表
    </h4>
     
    <div
      class="box-content"
    >
      <div
        class="search-input"
      >
        <input
          autocomplete="off"
          placeholder="模糊搜索"
          type="text"
        />
         
        <i
          class="input-cancel icon-cancel"
          style="display: none;"
        />
      </div>
       
      <div
        class="little-bar"
      >
        <div
          class="chosen-options"
        />
      </div>
       
      <div
        class="my-bar"
      >
        <ul>
          <li
            data-id="2"
            data-name="b"
          >
            <input
              id="2"
              type="checkbox"
            />
             
            <label
              for="2"
            >
              b2
            </label>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
